apps/tom-server/Dockerfile

```
@@ -0,0 +1,123 @@
+# =============================================================================
+# apps/tom-server Dockerfile
+# Nx-based multi-stage build
+# =============================================================================
+
+# -----------------------------------------------------------------------------
+# Stage 1: Builder - Install dependencies and build
+# -----------------------------------------------------------------------------
+FROM node:24.14.0-alpine AS builder
+
+# Update image and install build tools
+RUN apk update && apk upgrade && \
```

Hadolint (2.14.0)

[warning] Pin versions in apk add. Instead of `apk add <package>` use `apk add <package>=<version>`

(DL3018)